Remove `discovery.type` BWC layer from the EC2/Azure/GCE plugins

Those plugins don't replace the discovery logic but rather only provide a custom unicast host provider for their respective platforms. in 5.1 we introduced the  `discovery.zen.hosts_provider` setting to better reflect it. We should remove the [BWC code](https://github.com/elastic/elasticsearch/blob/master/plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryPlugin.java#L105) in those plugins.

Thanks Boaz,

After removing the override I get a test error.  I am not sure if the
modification is bad or I need to modify/remove the test.  Could you provide
me with guidance?



John

Testing started at 6:17 PM ...
6:17:17 PM: Executing external task 'check'...
:buildSrc:compileJava UP-TO-DATE
:buildSrc:compileGroovy UP-TO-DATE
:buildSrc:writeVersionProperties UP-TO-DATE
:buildSrc:processResources UP-TO-DATE
:buildSrc:classes UP-TO-DATE
:buildSrc:jar UP-TO-DATE
:buildSrc:assemble UP-TO-DATE
:buildSrc:compileTestJava UP-TO-DATE
:buildSrc:compileTestGroovy NO-SOURCE
:buildSrc:processTestResources NO-SOURCE
:buildSrc:testClasses UP-TO-DATE
:buildSrc:test NO-SOURCE
:buildSrc:check UP-TO-DATE
:buildSrc:build UP-TO-DATE
=======================================
Elasticsearch Build Hamster says Hello!
=======================================
  Gradle Version        : 3.5
  OS Info               : Linux 4.4.0-78-generic (amd64)
  JDK Version           : Oracle Corporation 1.8.0_111 [Java HotSpot(TM)
64-Bit Server VM 25.111-b14]
  JAVA_HOME             : /usr/lib/jvm/jdk1.8.0_111
The executable property on ForkOptions has been deprecated and is scheduled
to be removed in Gradle 5.0. Please use javaHome instead.
Could not find vagrant: A problem occurred starting process 'command
'vagrant''
secured hdfsFixture is unsupported, please install Vagrant 1.8.6+ to enable
:core:compileJava UP-TO-DATE
:core:generateModulesList UP-TO-DATE
:core:generatePluginsList UP-TO-DATE
:core:processResources UP-TO-DATE
:core:classes UP-TO-DATE
:core:jar UP-TO-DATE
:plugins:discovery-gce:compileJava
:plugins:discovery-gce:compileJava - is not incremental (e.g. outputs have
changed, no previous execution, etc.).
:plugins:discovery-gce:processResources NO-SOURCE
:plugins:discovery-gce:classes
:plugins:discovery-gce:copyCheckstyleConf UP-TO-DATE
:plugins:discovery-gce:checkstyleMain
:client:rest:compileJava UP-TO-DATE
:client:rest:processResources NO-SOURCE
:client:rest:classes UP-TO-DATE
:client:rest:jar UP-TO-DATE
:plugins:discovery-gce:copyPluginPropertiesTemplate
:plugins:discovery-gce:pluginProperties UP-TO-DATE
:test:framework:compileJava UP-TO-DATE
:test:framework:processResources UP-TO-DATE
:test:framework:classes UP-TO-DATE
:test:framework:jar UP-TO-DATE
:plugins:discovery-gce:compileTestJava
:plugins:discovery-gce:createKey
:plugins:discovery-gce:processTestResources
:plugins:discovery-gce:testClasses
:plugins:discovery-gce:checkstyleTest
:plugins:discovery-gce:forbiddenApisMain
:plugins:discovery-gce:forbiddenApisTest
:plugins:discovery-gce:forbiddenApis
:plugins:discovery-gce:generateNotice UP-TO-DATE
:plugins:discovery-gce:jar
:plugins:discovery-gce:bundlePlugin
:distribution:buildCoreNotice UP-TO-DATE
:modules:transport-netty4:compileJava UP-TO-DATE
:modules:transport-netty4:processResources NO-SOURCE
:modules:transport-netty4:classes UP-TO-DATE
:modules:transport-netty4:jar UP-TO-DATE
:modules:transport-netty4:copyPluginPropertiesTemplate
:modules:transport-netty4:pluginProperties UP-TO-DATE
:modules:transport-netty4:bundlePlugin UP-TO-DATE
:distribution:buildTransportModules UP-TO-DATE
:distribution:integ-test-zip:createPluginsDir
:distribution:tools:java-version-checker:compileJava UP-TO-DATE
:distribution:tools:java-version-checker:processResources NO-SOURCE
:distribution:tools:java-version-checker:classes UP-TO-DATE
:distribution:tools:java-version-checker:jar UP-TO-DATE
:distribution:tools:plugin-cli:compileJava UP-TO-DATE
:distribution:tools:plugin-cli:processResources NO-SOURCE
:distribution:tools:plugin-cli:classes UP-TO-DATE
:distribution:tools:plugin-cli:jar UP-TO-DATE
:distribution:integ-test-zip:buildZip UP-TO-DATE
:plugins:discovery-gce:integTestCluster#prepareCluster.cleanShared
:plugins:discovery-gce:integTestCluster#clean
:plugins:discovery-gce:integTestCluster#checkPrevious SKIPPED
:plugins:discovery-gce:integTestCluster#stopPrevious SKIPPED
:plugins:discovery-gce:integTestCluster#extract
:plugins:discovery-gce:integTestCluster#configure
:plugins:discovery-gce:integTestCluster#copyPlugins
:plugins:discovery-gce:integTestCluster#installDiscoveryGcePlugin
:plugins:discovery-gce:integTestCluster#start
:rest-api-spec:compileJava NO-SOURCE
:rest-api-spec:processResources UP-TO-DATE
:rest-api-spec:classes UP-TO-DATE
:rest-api-spec:jar UP-TO-DATE
:plugins:discovery-gce:copyRestSpec UP-TO-DATE
:plugins:discovery-gce:checkstyle
:plugins:discovery-gce:dependencyLicenses
:plugins:discovery-gce:forbiddenPatterns UP-TO-DATE
:plugins:discovery-gce:jarHell
:plugins:discovery-gce:licenseHeaders
:test:logger-usage:compileJava UP-TO-DATE
:test:logger-usage:processResources NO-SOURCE
:test:logger-usage:classes UP-TO-DATE
:test:logger-usage:jar UP-TO-DATE
:plugins:discovery-gce:loggerUsageCheck
:build-tools:compileJava UP-TO-DATE
:build-tools:compileGroovy UP-TO-DATE
:build-tools:writeVersionProperties UP-TO-DATE
:build-tools:processResources UP-TO-DATE
:build-tools:classes UP-TO-DATE
:build-tools:jar UP-TO-DATE
:plugins:discovery-gce:namingConventions
:plugins:discovery-gce:thirdPartyAudit UP-TO-DATE
:plugins:discovery-gce:precommit
:plugins:discovery-gce:test
==> Test Info: seed=3D8ACCBEE4BD7E73; jvms=3; suites=4
Suite: org.elasticsearch.discovery.gce.GceDiscoverTests
  1> [2017-05-30T22:17:34,644][INFO ][o.e.d.g.GceDiscoverTests ]
[testJoin]: before test
  1> [2017-05-30T22:17:34,647][INFO ][o.e.d.g.GceDiscoverTests ]
[GceDiscoverTests#testJoin]: setting up test
  1> [2017-05-30T22:17:34,680][INFO ][o.e.t.InternalTestCluster] Setup
InternalTestCluster
[SUITE-CHILD_VM=[0]-CLUSTER_SEED=[2818117216213349196]-HASH=[2887AC9BF9AD]-cluster]
with seed [271BF6AD57CF334C] using [0] dedicated masters, [2] (data) nodes
and [0] coord only nodes (min_master_nodes are [auto-managed])
  1> [2017-05-30T22:17:34,869][INFO ][o.e.n.Node               ] [node_s0]
initializing ...
  1> [2017-05-30T22:17:34,901][INFO ][o.e.e.NodeEnvironment    ] [node_s0]
using [1] data paths, mounts [[/ (/dev/sda1)]], net usable_space [80gb],
net total_space [204.2gb], types [ext4]
  1> [2017-05-30T22:17:34,901][INFO ][o.e.e.NodeEnvironment    ] [node_s0]
heap size [491mb], compressed ordinary object pointers [true]
  1> [2017-05-30T22:17:34,902][INFO ][o.e.n.Node               ] [node_s0]
node name [node_s0], node ID [qXhJ4YgcR4us48OJOcU2-g]
  1> [2017-05-30T22:17:34,902][INFO ][o.e.n.Node               ] [node_s0]
version[6.0.0-alpha2-SNAPSHOT], pid[20900],
build[d1318e4/2017-05-30T10:04:07.753Z], OS[Linux/4.4.0-78-generic/amd64],
JVM[Oracle Corporation/Java HotSpot(TM) 64-Bit Server
VM/1.8.0_111/25.111-b14]
  1> [2017-05-30T22:17:34,902][INFO ][o.e.n.Node               ] [node_s0]
JVM arguments [-ea, -esa, -Xmx512m, -Xms512m,
-XX:+HeapDumpOnOutOfMemoryError,
-XX:HeapDumpPath=/home/murphy/elastic/elasticsearch/plugins/discovery-gce/build/heapdump,
-Dtests.prefix=tests, -Dtests.seed=3D8ACCBEE4BD7E73,
-Dtests.security.manager=false, -Djava.io.tmpdir=./temp,
-Dtests.gradle=true, -Dtests.artifact=discovery-gce,
-Djava.awt.headless=true, -Djdk.io.permissionsUseCanonicalPath=true,
-Dtests.ifNoTests=fail, -Dtests.task=:plugins:discovery-gce:test,
-Djna.nosys=true, -Dtests.logger.level=WARN,
-Djunit4.childvm.cwd=/home/murphy/elastic/elasticsearch/plugins/discovery-gce/build/testrun/test/J0,
-Djunit4.childvm.id=0, -Djunit4.childvm.count=3]
  1> [2017-05-30T22:17:34,902][WARN ][o.e.n.Node               ] [node_s0]
version [6.0.0-alpha2-SNAPSHOT] is a pre-release version of Elasticsearch
and is not suitable for production
  1> [2017-05-30T22:17:34,912][INFO ][o.e.p.PluginsService     ] [node_s0]
no modules loaded
  1> [2017-05-30T22:17:34,913][INFO ][o.e.p.PluginsService     ] [node_s0]
loaded plugin [org.elasticsearch.discovery.gce.GceDiscoverTests$TestPlugin]
  1> [2017-05-30T22:17:34,913][INFO ][o.e.p.PluginsService     ] [node_s0]
loaded plugin [org.elasticsearch.plugin.discovery.gce.GceDiscoveryPlugin]
  1> [2017-05-30T22:17:34,913][INFO ][o.e.p.PluginsService     ] [node_s0]
loaded plugin [org.elasticsearch.test.ESIntegTestCase$TestSeedPlugin]
  1> [2017-05-30T22:17:34,913][INFO ][o.e.p.PluginsService     ] [node_s0]
loaded plugin [org.elasticsearch.test.discovery.TestZenDiscovery$TestPlugin]
  1> [2017-05-30T22:17:34,913][INFO ][o.e.p.PluginsService     ] [node_s0]
loaded plugin [org.elasticsearch.transport.MockTcpTransportPlugin]
  1> [2017-05-30T22:17:34,913][WARN ][o.e.d.p.d.g.GceDiscoveryPlugin] Using
discovery.type setting to set hosts provider is deprecated. Set
"discovery.zen.hosts_provider: gce" instead
  1> [2017-05-30T22:17:35,647][INFO ][o.e.d.g.GceDiscoverTests ]
[GceDiscoverTests#testJoin]: cleaning up after test
  1> [2017-05-30T22:17:35,656][INFO ][o.e.d.g.GceDiscoverTests ]
[testJoin]: after test
  2> REPRODUCE WITH: gradle :plugins:discovery-gce:test
-Dtests.seed=3D8ACCBEE4BD7E73
-Dtests.class=org.elasticsearch.discovery.gce.GceDiscoverTests
-Dtests.method="testJoin" -Dtests.security.manager=false
-Dtests.locale=ar-MA -Dtests.timezone=Africa/Nouakchott
  2> REPRODUCE WITH: gradle :plugins:discovery-gce:test
-Dtests.seed=3D8ACCBEE4BD7E73
-Dtests.class=org.elasticsearch.discovery.gce.GceDiscoverTests
-Dtests.method="testJoin" -Dtests.security.manager=false
-Dtests.locale=ar-MA -Dtests.timezone=Africa/Nouakchott
ERROR   1.08s J0 | GceDiscoverTests.testJoin <<< FAILURES!
   > Throwable #1: java.lang.IllegalArgumentException: Unknown discovery
type [gce]
   >     at
org.elasticsearch.discovery.DiscoveryModule.<init>(DiscoveryModule.java:102)
   >     at org.elasticsearch.node.Node.<init>(Node.java:438)
   >     at org.elasticsearch.node.MockNode.<init>(MockNode.java:60)
   >     at
org.elasticsearch.test.InternalTestCluster.buildNode(InternalTestCluster.java:608)
   >     at
org.elasticsearch.test.InternalTestCluster.reset(InternalTestCluster.java:1040)
   >     at
org.elasticsearch.test.InternalTestCluster.beforeTest(InternalTestCluster.java:978)
   >     at
org.elasticsearch.test.ESIntegTestCase.beforeInternal(ESIntegTestCase.java:373)
   >     at
org.elasticsearch.test.ESIntegTestCase.setupTestCluster(ESIntegTestCase.java:2044)
   >     at java.lang.Thread.run(Thread.java:745)Throwable #2:
java.lang.RuntimeException: java.nio.file.FileAlreadyExistsException:
/home/murphy/elastic/elasticsearch/plugins/discovery-gce/build/testrun/test/J0/temp/org.elasticsearch.discovery.gce.GceDiscoverTests_3D8ACCBEE4BD7E73-001/tempDir-002/0
   >     at
org.elasticsearch.discovery.gce.GceDiscoverTests.nodeSettings(GceDiscoverTests.java:90)
   >     at
org.elasticsearch.test.ESIntegTestCase$2.nodeSettings(ESIntegTestCase.java:1839)
   >     at
org.elasticsearch.test.InternalTestCluster.getSettings(InternalTestCluster.java:365)
   >     at
org.elasticsearch.test.InternalTestCluster.buildNode(InternalTestCluster.java:579)
   >     at
org.elasticsearch.test.InternalTestCluster.buildNode(InternalTestCluster.java:562)
   >     at
org.elasticsearch.test.InternalTestCluster.getOrBuildRandomNode(InternalTestCluster.java:468)
   >     at
org.elasticsearch.test.InternalTestCluster.client(InternalTestCluster.java:648)
   >     at
org.elasticsearch.test.ESIntegTestCase.client(ESIntegTestCase.java:641)
   >     at
org.elasticsearch.test.ESIntegTestCase.client(ESIntegTestCase.java:634)
   >     at
org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:553)
   >     at
org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2058)
   >     at java.lang.Thread.run(Thread.java:745)
   > Caused by: java.nio.file.FileAlreadyExistsException:
/home/murphy/elastic/elasticsearch/plugins/discovery-gce/build/testrun/test/J0/temp/org.elasticsearch.discovery.gce.GceDiscoverTests_3D8ACCBEE4BD7E73-001/tempDir-002/0
   >     at
sun.nio.fs.UnixException.translateToIOException(UnixException.java:88)
   >     at
sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
   >     at
sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
   >     at
sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
   >     at
org.apache.lucene.mockfile.FilterFileSystemProvider.createDirectory(FilterFileSystemProvider.java:132)
   >     at
org.apache.lucene.mockfile.FilterFileSystemProvider.createDirectory(FilterFileSystemProvider.java:132)
   >     at
org.apache.lucene.mockfile.FilterFileSystemProvider.createDirectory(FilterFileSystemProvider.java:132)
   >     at
org.apache.lucene.mockfile.FilterFileSystemProvider.createDirectory(FilterFileSystemProvider.java:132)
   >     at java.nio.file.Files.createDirectory(Files.java:674)
   >     at
org.elasticsearch.discovery.gce.GceDiscoverTests.nodeSettings(GceDiscoverTests.java:88)
   >     ... 45 more
  2> NOTE: leaving temporary files on disk at:
/home/murphy/elastic/elasticsearch/plugins/discovery-gce/build/testrun/test/J0/temp/org.elasticsearch.discovery.gce.GceDiscoverTests_3D8ACCBEE4BD7E73-001
  2> NOTE: test params are: codec=Asserting(Lucene70): {}, docValues:{},
maxPointsInLeafNode=1020, maxMBSortInHeap=6.475715560444328,
sim=RandomSimilarity(queryNorm=false): {}, locale=ar-MA,
timezone=Africa/Nouakchott
  2> NOTE: Linux 4.4.0-78-generic amd64/Oracle Corporation 1.8.0_111
(64-bit)/cpus=4,threads=1,free=473409288,total=514850816
  2> NOTE: All tests run in this JVM: [GceDiscoverTests]
Completed [3/4] on J0 in 2.98s, 1 test, 1 error <<< FAILURES!
Tests with failures:
  - org.elasticsearch.discovery.gce.GceDiscoverTests.testJoin
JUnit4 test failed, ant output was:
   [junit4] <JUnit4> says kaixo! Master seed: 3D8ACCBEE4BD7E73
   [junit4] JVM J0:     0.67 ..     4.72 =     4.05s
   [junit4] JVM J1:     0.37 ..     3.42 =     3.05s
   [junit4] JVM J2:     0.37 ..     2.95 =     2.59s
   [junit4] Execution time total: 4.73 sec.
   [junit4] Tests summary: 4 suites, 22 tests, 1 error
:plugins:discovery-gce:test FAILED
:plugins:discovery-gce:integTestCluster#stop
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':plugins:discovery-gce:test'.
> There were test failures: 4 suites, 22 tests, 1 error [seed:
3D8ACCBEE4BD7E73]
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or
--debug option to get more log output.
BUILD FAILED
Total time: 18.286 secs
There were test failures: 4 suites, 22 tests, 1 error [seed:
3D8ACCBEE4BD7E73]


On Tue, May 30, 2017 at 7:29 AM, Boaz Leskes <notifications@github.com>
wrote:

> Thanks for picking this up. Let us know if you need any help.
>
> > On 30 May 2017, at 12:35, John Murphy <notifications@github.com> wrote:
> >
> > I am still working on this. I have to learn a new build tool (gradle), a
> > new IDE (intelliJ) and a new code base(Elasticsearch), but I will
> deliver.
> >
> >
> > On Sun, May 28, 2017 at 1:45 AM, Ryan Ernst <notifications@github.com>
> > wrote:
> >
> > > I believe some of the overridden additionalSettings() can also be
> > > removed, which injected the appropriate new setting when detecting the
> old
> > > setting.
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/elastic/elasticsearch/issues/24543#
> issuecomment-304494396>,
> > > or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/AJQUxyQAHR5Ci76s_
> oyLdN7kf-kx67Cuks5r-QnvgaJpZM4NTvJ5>
> > > .
> > >
> > —
> > You are receiving this because you authored the thread.
>
> > Reply to this email directly, view it on GitHub, or mute the thread.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/elasticsearch/issues/24543#issuecomment-304850109>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJQUxxf1vssCDrXu_2FZLSINM8otMgDvks5r-_2GgaJpZM4NTvJ5>
> .
>

--- a/plugins/discovery-azure-classic/src/main/java/org/elasticsearch/plugin/discovery/azure/classic/AzureDiscoveryPlugin.java
+++ b/plugins/discovery-azure-classic/src/main/java/org/elasticsearch/plugin/discovery/azure/classic/AzureDiscoveryPlugin.java
@@ -22,24 +22,15 @@ package org.elasticsearch.plugin.discovery.azure.classic;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.cloud.azure.classic.management.AzureComputeService;
 import org.elasticsearch.cloud.azure.classic.management.AzureComputeServiceImpl;
-import org.elasticsearch.cluster.routing.allocation.AllocationService;
-import org.elasticsearch.cluster.service.ClusterApplier;
-import org.elasticsearch.cluster.service.MasterService;
-import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
 import org.elasticsearch.common.logging.DeprecationLogger;
 import org.elasticsearch.common.logging.Loggers;
 import org.elasticsearch.common.network.NetworkService;
-import org.elasticsearch.common.settings.ClusterSettings;
 import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.discovery.Discovery;
-import org.elasticsearch.discovery.DiscoveryModule;
 import org.elasticsearch.discovery.azure.classic.AzureUnicastHostsProvider;
 import org.elasticsearch.discovery.zen.UnicastHostsProvider;
-import org.elasticsearch.discovery.zen.ZenDiscovery;
 import org.elasticsearch.plugins.DiscoveryPlugin;
 import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.threadpool.ThreadPool;
 import org.elasticsearch.transport.TransportService;
 
 import java.util.Arrays;
@@ -73,17 +64,7 @@ public class AzureDiscoveryPlugin extends Plugin implements DiscoveryPlugin {
             () -> new AzureUnicastHostsProvider(settings, createComputeService(), transportService, networkService));
     }
 
-    @Override
-    public Map<String, Supplier<Discovery>> getDiscoveryTypes(ThreadPool threadPool, TransportService transportService,
-                                                              NamedWriteableRegistry namedWriteableRegistry,
-                                                              MasterService masterService, ClusterApplier clusterApplier,
-                                                              ClusterSettings clusterSettings, UnicastHostsProvider hostsProvider,
-                                                              AllocationService allocationService) {
-        // this is for backcompat with pre 5.1, where users would set discovery.type to use ec2 hosts provider
-        return Collections.singletonMap(AZURE, () ->
-            new ZenDiscovery(settings, threadPool, transportService, namedWriteableRegistry, masterService, clusterApplier,
-                clusterSettings, hostsProvider, allocationService));
-    }
+
 
     @Override
     public List<Setting<?>> getSettings() {
@@ -99,19 +80,5 @@ public class AzureDiscoveryPlugin extends Plugin implements DiscoveryPlugin {
                             AzureComputeService.Discovery.ENDPOINT_NAME_SETTING);
     }
 
-    @Override
-    public Settings additionalSettings() {
-        // For 5.0, the hosts provider was "zen", but this was before the discovery.zen.hosts_provider
-        // setting existed. This check looks for the legacy setting, and sets hosts provider if set
-        String discoveryType = DiscoveryModule.DISCOVERY_TYPE_SETTING.get(settings);
-        if (discoveryType.equals(AZURE)) {
-            deprecationLogger.deprecated("using [" + DiscoveryModule.DISCOVERY_TYPE_SETTING.getKey() +
-                "] to set hosts provider is deprecated; " +
-                "set \"" + DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING.getKey() + ": " + AZURE + "\" instead");
-            if (DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING.exists(settings) == false) {
-                return Settings.builder().put(DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING.getKey(), AZURE).build();
-            }
-        }
-        return Settings.EMPTY;
-    }
+
 }

--- a/plugins/discovery-azure-classic/src/test/java/org/elasticsearch/cloud/azure/classic/AbstractAzureComputeServiceTestCase.java
+++ b/plugins/discovery-azure-classic/src/test/java/org/elasticsearch/cloud/azure/classic/AbstractAzureComputeServiceTestCase.java
@@ -23,8 +23,6 @@ import org.elasticsearch.action.admin.cluster.node.info.NodesInfoResponse;
 import org.elasticsearch.cloud.azure.classic.management.AzureComputeService.Discovery;
 import org.elasticsearch.cloud.azure.classic.management.AzureComputeService.Management;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.node.Node;
-import org.elasticsearch.plugin.discovery.azure.classic.AzureDiscoveryPlugin;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.test.ESIntegTestCase;
 
@@ -44,7 +42,7 @@ public abstract class AbstractAzureComputeServiceTestCase extends ESIntegTestCas
     protected Settings nodeSettings(int nodeOrdinal) {
         Settings.Builder builder = Settings.builder()
             .put(super.nodeSettings(nodeOrdinal))
-            .put("discovery.type", "azure");
+            .put("discovery.zen.hosts_provider", "azure");
 
         // We add a fake subscription_id to start mock compute service
         builder.put(Management.SUBSCRIPTION_ID_SETTING.getKey(), "fake")

--- a/plugins/discovery-azure-classic/src/test/java/org/elasticsearch/discovery/azure/classic/AzureDiscoveryClusterFormationTests.java
+++ b/plugins/discovery-azure-classic/src/test/java/org/elasticsearch/discovery/azure/classic/AzureDiscoveryClusterFormationTests.java
@@ -106,7 +106,7 @@ public class AzureDiscoveryClusterFormationTests extends ESIntegTestCase {
             throw new RuntimeException(e);
         }
         return Settings.builder().put(super.nodeSettings(nodeOrdinal))
-            .put(DiscoveryModule.DISCOVERY_TYPE_SETTING.getKey(), AzureDiscoveryPlugin.AZURE)
+            .put(DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING.getKey(), AzureDiscoveryPlugin.AZURE)
             .put(Environment.PATH_LOGS_SETTING.getKey(), resolve)
             .put(TransportSettings.PORT.getKey(), 0)
             .put(Node.WRITE_PORTS_FILE_SETTING.getKey(), "true")

--- a/plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryPlugin.java
+++ b/plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryPlugin.java
@@ -24,33 +24,24 @@ import org.apache.logging.log4j.Logger;
 import org.apache.lucene.util.IOUtils;
 import org.apache.lucene.util.SetOnce;
 import org.elasticsearch.SpecialPermission;
-import org.elasticsearch.cluster.routing.allocation.AllocationService;
-import org.elasticsearch.cluster.service.ClusterApplier;
 import org.elasticsearch.common.SuppressForbidden;
-import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
 import org.elasticsearch.common.logging.DeprecationLogger;
 import org.elasticsearch.common.logging.Loggers;
 import org.elasticsearch.common.network.NetworkService;
-import org.elasticsearch.common.settings.ClusterSettings;
 import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.discovery.Discovery;
-import org.elasticsearch.discovery.DiscoveryModule;
-import org.elasticsearch.cluster.service.MasterService;
 import org.elasticsearch.discovery.zen.UnicastHostsProvider;
-import org.elasticsearch.discovery.zen.ZenDiscovery;
 import org.elasticsearch.node.Node;
 import org.elasticsearch.plugins.DiscoveryPlugin;
 import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.threadpool.ThreadPool;
 import org.elasticsearch.transport.TransportService;
 
+import java.io.UncheckedIOException;
 import java.io.BufferedReader;
-import java.io.Closeable;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
-import java.io.UncheckedIOException;
+import java.io.Closeable;
 import java.net.URL;
 import java.net.URLConnection;
 import java.nio.charset.StandardCharsets;
@@ -95,17 +86,7 @@ public class Ec2DiscoveryPlugin extends Plugin implements DiscoveryPlugin, Close
         this.settings = settings;
     }
 
-    @Override
-    public Map<String, Supplier<Discovery>> getDiscoveryTypes(ThreadPool threadPool, TransportService transportService,
-                                                              NamedWriteableRegistry namedWriteableRegistry,
-                                                              MasterService masterService, ClusterApplier clusterApplier,
-                                                              ClusterSettings clusterSettings, UnicastHostsProvider hostsProvider,
-                                                              AllocationService allocationService) {
-        // this is for backcompat with pre 5.1, where users would set discovery.type to use ec2 hosts provider
-        return Collections.singletonMap(EC2, () ->
-            new ZenDiscovery(settings, threadPool, transportService, namedWriteableRegistry, masterService, clusterApplier,
-                clusterSettings, hostsProvider, allocationService));
-    }
+
 
     @Override
     public NetworkService.CustomNameResolver getCustomNameResolver(Settings settings) {

--- a/plugins/discovery-gce/src/main/java/org/elasticsearch/plugin/discovery/gce/GceDiscoveryPlugin.java
+++ b/plugins/discovery-gce/src/main/java/org/elasticsearch/plugin/discovery/gce/GceDiscoveryPlugin.java
@@ -29,24 +29,15 @@ import org.elasticsearch.cloud.gce.GceInstancesServiceImpl;
 import org.elasticsearch.cloud.gce.GceMetadataService;
 import org.elasticsearch.cloud.gce.network.GceNameResolver;
 import org.elasticsearch.cloud.gce.util.Access;
-import org.elasticsearch.cluster.routing.allocation.AllocationService;
-import org.elasticsearch.cluster.service.ClusterApplier;
-import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
 import org.elasticsearch.common.logging.DeprecationLogger;
 import org.elasticsearch.common.logging.Loggers;
 import org.elasticsearch.common.network.NetworkService;
-import org.elasticsearch.common.settings.ClusterSettings;
 import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.discovery.Discovery;
-import org.elasticsearch.discovery.DiscoveryModule;
-import org.elasticsearch.cluster.service.MasterService;
 import org.elasticsearch.discovery.gce.GceUnicastHostsProvider;
 import org.elasticsearch.discovery.zen.UnicastHostsProvider;
-import org.elasticsearch.discovery.zen.ZenDiscovery;
 import org.elasticsearch.plugins.DiscoveryPlugin;
 import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.threadpool.ThreadPool;
 import org.elasticsearch.transport.TransportService;
 
 import java.io.Closeable;
@@ -83,17 +74,7 @@ public class GceDiscoveryPlugin extends Plugin implements DiscoveryPlugin, Close
         logger.trace("starting gce discovery plugin...");
     }
 
-    @Override
-    public Map<String, Supplier<Discovery>> getDiscoveryTypes(ThreadPool threadPool, TransportService transportService,
-                                                              NamedWriteableRegistry namedWriteableRegistry,
-                                                              MasterService masterService, ClusterApplier clusterApplier,
-                                                              ClusterSettings clusterSettings, UnicastHostsProvider hostsProvider,
-                                                              AllocationService allocationService) {
-        // this is for backcompat with pre 5.1, where users would set discovery.type to use ec2 hosts provider
-        return Collections.singletonMap(GCE, () ->
-            new ZenDiscovery(settings, threadPool, transportService, namedWriteableRegistry, masterService, clusterApplier,
-                clusterSettings, hostsProvider, allocationService));
-    }
+
 
     @Override
     public Map<String, Supplier<UnicastHostsProvider>> getZenHostsProviders(TransportService transportService,
@@ -122,21 +103,7 @@ public class GceDiscoveryPlugin extends Plugin implements DiscoveryPlugin, Close
             GceInstancesService.MAX_WAIT_SETTING);
     }
 
-    @Override
-    public Settings additionalSettings() {
-        // For 5.0, the hosts provider was "zen", but this was before the discovery.zen.hosts_provider
-        // setting existed. This check looks for the legacy setting, and sets hosts provider if set
-        String discoveryType = DiscoveryModule.DISCOVERY_TYPE_SETTING.get(settings);
-        if (discoveryType.equals(GCE)) {
-            deprecationLogger.deprecated("Using " + DiscoveryModule.DISCOVERY_TYPE_SETTING.getKey() +
-                " setting to set hosts provider is deprecated. " +
-                "Set \"" + DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING.getKey() + ": " + GCE + "\" instead");
-            if (DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING.exists(settings) == false) {
-                return Settings.builder().put(DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING.getKey(), GCE).build();
-            }
-        }
-        return Settings.EMPTY;
-    }
+
 
     @Override
     public void close() throws IOException {

--- a/plugins/discovery-gce/src/test/java/org/elasticsearch/discovery/gce/GceDiscoverTests.java
+++ b/plugins/discovery-gce/src/test/java/org/elasticsearch/discovery/gce/GceDiscoverTests.java
@@ -90,7 +90,7 @@ public class GceDiscoverTests extends ESIntegTestCase {
             throw new RuntimeException(e);
         }
         return Settings.builder().put(super.nodeSettings(nodeOrdinal))
-            .put("discovery.type", "gce")
+            .put("discovery.zen.hosts_provider", "gce")
             .put("path.logs", resolve)
             .put("transport.tcp.port", 0)
             .put("node.portsfile", "true")

